### PR TITLE
[EDCO-19] Footer Component

### PR DESCRIPTION
**Pull request by leleOFA**
Footer Component

### CODE REVIEW

File: src/components/Footer/Footer.tsx [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+import { hrefNoOp, wrapHandleExitAction } from '../../utils/index';
+
+/* Icons */
+import TwitterIcon from '@mui/icons-material/Twitter';
+import LinkedIcon from '@mui/icons-material/LinkedIn';
+import InstagramIcon from '@mui/icons-material/Instagram';
+/* import { MediumIcon } from '@icons/MediumIcon';
+import { LogoPagoPACompany } from '@assets/LogoPagoPACompany';
+import { FundedByNextGenerationEU } from '@assets/FundedByNextGenerationEU'; */
+
```
Comment (stratoivandiluccio) [inline src/components/Footer/Footer.tsx:20]: please remove these commented imports.

File: src/components/Footer/Footer.tsx [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+                  <Typography variant="overline">Prodotti e Servizi</Typography>
+                )}
+
+                <Stack
+                  component="ul"
+                  alignItems={{ xs: 'center', sm: 'start' }}
+                  textAlign="center"
+                  sx={{ padding: 0, listStyle: 'none' }}
+                >
+                  {jsonProducts &&
```
Comment (mmoio) [inline src/components/Footer/Footer.tsx:154]: You can remove the jsonProducts &&, leaving only jsonProducts?.map and remove the disable rule on top of the file.

File: src/components/Footer/Footer.tsx [hunk 3]
```diff
@@ -0,0 +1,317 @@
+/* eslint-disable @typescript-eslint/prefer-optional-chain */
```
Comment (mmoio) [inline src/components/Footer/Footer.tsx:1]: You can remove this eslint-disable and correct line 154, see comment below.

File: src/components/Footer/index.ts [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+}
+
+export class EnumType<A> extends t.Type<A> {
+  public readonly _tag = 'EnumType' as const;
+  public enumObject!: object;
+  public constructor(e: object, name?: string) {
+    super(
+      name ?? 'enum',
+      (u): u is A => {
+        if (!Object.values(this.enumObject).find((v) => v === u)) {
```
Comment (mmoio) [inline src/components/Footer/index.ts:55]: You can use .includes(u) instead of find

File: src/components/Footer/index.ts [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+}
+
+export interface CompanyLinkType {
+  /** the url to witch the user will be redirect */
+  href?: string;
+  ariaLabel: string;
+  /** if defined it will override the href behavior */
+  onClick?: () => void;
+}
+
```
Comment (mmoio) [inline src/components/Footer/index.ts:47]: Can you add a comment explaining what this class does and why do you need it?

File: src/components/Footer/Footer.tsx [hunk 6]
```diff
@@ -0,0 +1,317 @@
...
+import InstagramIcon from '@mui/icons-material/Instagram';
+
+/* Enum */
+import { ProductArrayType } from './ProductType';
+import { FundedByNextGenerationEU } from '../../assets/FundedByNextGenerationEU';
+import { LogoPagoPACompany } from '../../assets/LogoPagoPACompany';
+
+type FooterProps = LangSwitchProps & {
+  companyLink: CompanyLinkType;
+  legalInfo: JSX.Element | JSX.Element[];
```
Comment (mmoio) [inline src/components/Footer/Footer.tsx:25]: Replace JSX.Element with the Generic type defined inside types/components

File: src/utils/index.ts [hunk 7]
```diff
@@ -7,3 +7,27 @@ import { type Generic } from 'types/components';
...
+          (() => {
+            window.location.assign(href);
+          })
+      );
+    } else if (onClick) {
+      onClick();
+    } else {
+      window.location.assign(href);
+    }
+  };
```
Comment (mmoio) [inline src/utils/index.ts:31]: You can change line 19 to 30 to something like:
```js
    const exitAction =
      onClick ??
      (() => {
        window.location.assign(href);
      });
    if (onExit) onExit(exitAction);
    else exitAction();
```

File: src/components/Footer/Footer.tsx (hunk 1 above)
Comment (leleOFA) [reply src/components/Footer/Footer.tsx:17]: do it

File: src/components/Footer/Footer.tsx (hunk 2 above)
Comment (leleOFA) [reply src/components/Footer/Footer.tsx:154]: do it

File: src/components/Footer/Footer.tsx (hunk 3 above)
Comment (leleOFA) [reply src/components/Footer/Footer.tsx:1]: do it

File: src/components/Footer/index.ts [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+      (u): u is A => {
+        if (!Object.values(this.enumObject).find((v) => v === u)) {
+          return false;
+        }
+        // enum reverse mapping check
+        if (typeof (this.enumObject as any)[u as string] === 'number') {
+          return false;
+        }
+        return true;
+      },
```
[the remark this answers is not in the document]
Comment (leleOFA) [reply src/components/Footer/index.ts:63]: do it

File: src/components/Footer/index.ts (hunk 4 above)
Comment (leleOFA) [reply src/components/Footer/index.ts:55]: change from previous comment

File: src/components/Footer/index.ts (hunk 5 above)
Comment (leleOFA) [reply src/components/Footer/index.ts:47]: in case you need to add a function

File: src/components/Footer/Footer.tsx (hunk 6 above)
Comment (leleOFA) [reply src/components/Footer/Footer.tsx:25]: done

File: src/utils/index.ts (hunk 7 above)
Comment (leleOFA) [reply src/utils/index.ts:31]: done

**Review by mmoio — APPROVED** [review]

**Review by stratoivandiluccio — APPROVED** [review]

**Pull request merged**